templates/community/community.js

```diff
@@ -0,0 +1,304 @@
+import {
+  buildBlock, decorateBlock, getMetadata, loadBlock,
+} from '../../scripts/aem.js';
+import {
+  a,
+  aside,
+  br,
+  div,
+  form,
+  h1,
+  h2,
+  h3,
+  h4,
+  option,
+  select,
+} from '../../scripts/dom-helpers.js';
+import {
+  filters,
+  getHeaderTitleForFilter,
+  getInventoryHomes,
+} from '../../scripts/inventory.js';
+// import { getHomePlans } from '../../scripts/home-plans-data.js';
+import { getSalesCenters } from '../../scripts/sales-center.js';
+import { loadTemplateBlock } from '../../scripts/template-block.js';
+import { getCommunityDetails } from '../../scripts/communities.js';
+import { createActionBar } from '../../scripts/block-helper.js';
+
+/**
+ * Builds the inventory homes block.
+ *
+ * @returns {Promise<Element>} The inventory homes block wrapped in a div.
+ */
+async function buildInventoryHomes() {
+  const modelsBlock = buildBlock('models', []);
+  modelsBlock.classList.add('inventory');
+  const blockWrapper = div(modelsBlock);
+  decorateBlock(modelsBlock);
+  await loadBlock(modelsBlock);
+  return blockWrapper;
+}
+
+async function fetchRequiredPageData(filter) {
+  // These three calls could be done differently, but for now, we will keep them separate.
+  // It might be nice to have a factory that takes the results of these calls and builds the
+  // required data for the page.
+  // For example this page could ask for the 3 sheets at once and then build the required data.
+  const salesCenterData = await getSalesCenters(window.location);
+  const community = await getCommunityDetails(window.location.pathname);
+  const homes = await getInventoryHomes(community.name, filter);
+
+  window.hh = window.hh || {};
+  window.hh.current = {};
+  window.hh.current.models = homes;
+  window.hh.current.sale_center = salesCenterData.sales_center;
+  return {
+    salesCenter: salesCenterData.sales_center,
+    community,
+  };
+}
+
+function DeferredPromise() {
+  const deferred = {};
+
+  deferred.promise = new Promise((resolve, reject) => {
+    deferred.resolve = resolve;
+    deferred.reject = reject;
+  });
+
+  return deferred;
+}
+
+async function createSpecialists(specialists) {
+  const agents = [];
+  const deferred = DeferredPromise();
+  const promises = [];
+
+  specialists.forEach((specialist) => {
+    const content = [];
+    content.push(['name', specialist.name]);
+    content.push(['phone', specialist.phone]);
+    content.push(['photo', specialist.headshotImage]);
+    content.push(['email', specialist.email]);
+
+    const specialistsBlock = buildBlock('specialists', content);
```

Comment:
Wouldn't it be better to have the buildBlock and decorateBlock functions in templates/blocks/specialists.js and return the block element already decorated? This way, the block creation code is tucked away in its file, and the caller of the function appends the block to the template.